odd_collector_azure/adapters/azure_data_factory/adapter.py

```
@@ -0,0 +1,86 @@
+            )
+            for pipeline in pipelines:
+                activities_entities_tmp = []
+                self.generator.set_oddrn_paths(pipelines=pipeline.name)
+                pipelines_runs = self.client.get_pipeline_runs(pipeline.name)
+                pipelines_runs_entities.extend(
+                    [map_pipeline_run(self.generator, run) for run in pipelines_runs]
+                )
+
+                activities = [
+                    ADFActivity(act, all_activities=pipeline.activities)
+                    for act in pipeline.activities
+                ]
+                activities_runs = self.client.get_activity_runs(pipeline.name)
+
+                for activity in activities:
+                    self.generator.set_oddrn_paths(activities=activity.name)
+                    runs = activities_runs[activity.name]
+                    activities_entities_tmp.extend(
+                        map_activity(self.generator, activity)
+                    )
+                    activities_runs_entities.extend(
+                        [map_activity_run(self.generator, run) for run in runs]
+                    )
+                pipelines_entities.append(
+                    map_pipeline(self.generator, pipeline, activities_entities_tmp)
+                )
+                activities_entities.extend(activities_entities_tmp)
+
+            factory_entity = map_factory(self.generator, factory, pipelines_entities)
+
+            return DataEntityList(
+                data_source_oddrn=self.get_data_source_oddrn(),
+                items=[
+                    *activities_runs_entities,
+                    *activities_entities,
+                    *pipelines_runs_entities,
+                    *pipelines_entities,
+                    factory_entity,
+                ],
+            )
+        except Exception as e:
+            logger.error(f"Error while processing: {e}." " SKIPPING.", exc_info=True)
```

Re-raise an exception, collector should handle it. Or just not use `try/catch`.
Client and mapping functions should raise a specific exception, i.e
`from odd_collector_sdk.errors import MappingDataError, DataSourceConnectionError`